rwlib/source/rw/bit_cast.cpp

```diff
@@ -0,0 +1,19 @@
+#ifndef _LIBRW_BIT_CAST_CPP_
+#define _LIBRW_BIT_CAST_CPP_
+
+//Based on https://gist.github.com/socantre/3472964
+#include <cstring> // memcpy
+#include <type_traits> // is_trivially_copyable
+#include "rw/defines.hpp" // RW_ASSERT
+
+template <class Dest, class Source>
+inline Dest bit_cast(Source const &source) {
+    RW_ASSERT(std::is_trivially_copyable<Dest>::value);
```

Comment:
You can make these ``static_assert``, since they're know at compile time.

Reply:
I'll change.